extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -465,23 +491,48 @@ export class AmpLightboxViewer extends AMP.BaseElement {
     this.updateInViewport(dev().assertElement(this.container_), true);
     this.scheduleLayout(dev().assertElement(this.container_));
 
-    this.currentElementId_ = element.lightboxItemId;
-    // Hack to access private property. Better than not getting
-    // type checking to work.
-    /**@type {?}*/ (this.carousel_).implementation_.showSlideWhenReady(
-        this.currentElementId_);
-
     this.win.document.documentElement.addEventListener(
         'keydown', this.boundHandleKeyboardEvents_);
 
-    return Promise.resolve();
+    return this.resources_.requireLayout(this.carousel_)
+        .then(() => {
+          this.currentElemId_ = element.lightboxItemId;
+          /**@type {?}*/ (this.carousel_).implementation_
```

Comment:
can this become just `showSlide` now given you wait for layout?

Reply:
The public function exposed by `amp-carousel` is actually `showSlideWhenReady`. `showSlide` is a private function.